packages/orbit-components/src/InputField/index.js

```diff
@@ -329,10 +331,10 @@ const StyledIconWrapper = styled.span`
   display: flex;
 `;
 
-const InputField: React.AbstractComponent<Props, HTMLInputElement> = React.forwardRef<
+const InputField: React.AbstractComponent<
   Props,
-  HTMLInputElement,
->((props, ref) => {
+  React.AbstractComponent<Props, HTMLInputElement>,
+> = React.forwardRef((props, ref) => {
```

Comment:
I'm not sure if this is right, it should probably be just `React.AbstractComponent<Props, HTMLInputElement>`, Flow doesn't seem to complain. You can simply import the type exported by `InputField/index.js.flow`:

```js
import typeof InputFieldType from ".";

// ...

const InputField: InputFieldType = // ...
```

That way you also check whether the library definition is good.

Reply:
it fails for, types are correct, its actually two times forwarding refs, so the target element in the first type is not `HTMLInputElement` but another `AbstractComponent`